fe/fe-core/src/main/java/com/starrocks/analysis/ColumnDef.java

```diff
@@ -86,6 +86,7 @@ public DefaultValueDef(boolean isSet, Expr expr) {
     private final TypeDef typeDef;
     private AggregateType aggregateType;
     private boolean isKey;
+    private Boolean isAllowNullTristate;
```

Comment:
I prefer a enum to describe this.

Reply:
do you mean the variable isPrimaryKeyAllowNull should be a enum? 
or mean that the method isAllowNull() should return a enum, and do the judgement logic for primary key every single places?